code/modules/reagents/Chemistry-Goon-Toxins.dm

```diff
@@ -204,4 +204,169 @@ datum/reagent/itching_powder/on_mob_life(var/mob/living/M as mob)
 	result = "facid"
 	required_reagents = list("sacid" = 1, "fluorine" = 1, "hydrogen" = 1, "potassium" = 1)
 	result_amount = 4
-	required_temp = 380
\ No newline at end of file
+	required_temp = 380
+
+datum/reagent/regadenoson
+	name = "Regadenoson"
+	id = "regadenoson"
+	description = "Major stamina regeneration buff. 33% chance to hit with 5-25 TOX. 5-10% chances to stun and cause suffocation or immediate heart failure."
+	reagent_state = LIQUID
+	color = "#CF3600" // rgb: 207, 54, 0
+	metabolization_rate = 0.4
+
+datum/reagent/regadenoson/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustStaminaLoss(-5)
+	if(prob(33))
+		M.adjustToxLoss(rand(5,25))
+	if(prob(rand(5,10)))
+		var/picked_option = rand(1,3)
+		switch(picked_option)
+			if(1)
+				M.Stun(3)
+				M.Weaken(3)
+			if(2)
+				M.losebreath += 10
+				M.adjustOxyLoss(rand(5,25))
+			if(3)
+				M.visible_message("<span class = 'userdanger'>[M] clutches at their chest as if their heart stopped!</span>")
+				M.adjustBruteLoss(100) // rip in pepperoni
```

Comment:
If you guys had internal organs like Bay (hey, you have blood, that's a start), this would be easy...but since you don't....well...it's hard to test Initro, quite frankly, because it tends to uh..well...kill you; in any event, when you DO have a heart attack, you start accumulating losebreath and brain damage over time; once it starts, you have about 2-4 seconds before you black out from it.